privacyidea/lib/policy.py

```diff
@@ -809,6 +817,32 @@ def _policy_matches_request_header_condition(policy, key, comparator, value, req
             raise PolicyError(u"Policy {!r} has conditions on headers {!r}, but http header"
                         u" is not available".format(policy["name"], key))
 
+    @staticmethod
+    def _policy_matches_token_condition(policy, key, comparator, value, db_token):
+        """
+
+        :param policy: a policy dictionary, the policy in question
+        :param key: the column name of the token
+        :param comparator: a value comparator: one of "equal", "contains"
+        :param value: a value against which the token value will be compared
+        :param db_token: a dbtoken object
+        :return: bool
+        """
+        if db_token and key in db_token.get():
+            try:
+                return compare_values(db_token.get(key), comparator, value)
+            except Exception as exx:
+                log.warning(u"Error during handling the condition on token {!r} "
+                            u"of policy {!r}: {!r}".format(key, policy['name'], exx))
+                raise PolicyError(
+                    u"Invalid comparison in the {!s} conditions of policy {!r}".format(type, policy['name']))
+        else:
```

Comment:
This else is reached if "not db_token OR key not in dbtoken", so also if we have _no_ dbtoken. This happens e.g. at /auth if any matching admin policy has a token condition. 
No login is possible and the error message is misleading, since the reason is that there is simply no token object available.

Please check in this else if there is a dbtoken and if not throw something similar to

"Policy {!r} has condition on token, but an according token object is not available"

as in `_policy_matches_info_condition`

Reply:
I will change this to the flow of the other extended conditions.
However, in certain cases login will still not be possible (as it is also for other conditions.) This is simply a logical problem, which I will not address in the PR, which only deals with the token attributes. (I opened #2676 to do this)

Reply:
done in 0f0588ce